packages/core/test/errors.test.ts

```diff
@@ -892,4 +893,41 @@ describe('error handling', () => {
       error_thrown_in_guard_when_transitioning]
     `);
   });
+
+  it('error thrown when resolving the initial context should rethrow synchronously', (done) => {
+    const machine = createMachine({
+      context: () => {
+        throw new Error('oh no');
+      }
+    });
+
+    const actor = createActor(machine);
+
+    installGlobalOnErrorHandler(() => {
+      done.fail();
+    });
+
+    expect(() => actor.start()).toThrowErrorMatchingInlineSnapshot(`"oh no"`);
```

Comment:
actors are always async conceptually, a synchronous throw is something that might sound convenient at first but you essentially create 2 error channels by introducing this. Now, if you want to handle errors, u need to both install an error listener and wrap this in try/catch. The guiding principle that we had when designing this was that you should always be notified about errors in the very same way (through error listeners).

Reply:
@Andarist In theory I agree, but in practice, if we know that an actor is going to fail right at the start, rather than later during event processing, it's a _much_ better DX (IMO) to throw right away.

Reply:
This is my feeling too.  User experience > theoretical purity

Reply:
@boneskull I may have changed my stance on this: https://www.youtube.com/watch?v=JCXZhe6KsxQ&t=294s

Reply:
I don't follow

Reply:
@davidkpiano Should I close this?

Reply:
Yes, let's close it for now, given that it's a breaking change.